File: dsa_algo/leetcode/arrays/lc/java/mergeSortedArrs.java
public class mergeSortedArrs{
    public void merge(int[] nums1, int m, int[] nums2, int n) {
       int p1 = m - 1;
       int p2 = n - 1;
       for(int insert_loc = m + n - 1; insert_loc >= 0; insert_loc--) 
       {
            if(p2 < 0)
                break;
            if(p1 >= 0 && nums1[p1] > nums2[p2])
                nums1[insert_loc] = nums1[p1--];
            else
                nums1[insert_loc] = nums2[p2--];
       }
    }
}

File: dsa_algo/leetcode/arrays/lc/python/merge_sorted_arrs.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        p1 = m - 1
        p2 = n - 1
        for insert_loc in range(n + m - 1, -1, -1):
            if p2 < 0:
                break
            if p1 >= 0 and nums1[p1] > nums2[p2]:
                nums1[insert_loc] = nums1[p1]
                p1 -= 1
            else:
                nums1[insert_loc] = nums2[p2]
                p2 -= 1
